code/site/components/com_pages/data/factory.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Joomlatools Pages
+ *
+ * @copyright   Copyright (C) 2018 Johan Janssens and Timble CVBA. (http://www.timble.net)
+ * @license     GNU GPLv3 <http://www.gnu.org/licenses/gpl.html>
+ * @link        https://github.com/joomlatools/joomlatools-framework-pages for the canonical source repository
+ */
+
+final class ComPagesDataFactory extends KObject implements KObjectSingleton
+{
+    /**
+     * The data object cache
+     *
+     * @var string
+     */
+    private static $__cache = array();
+
+    /**
+     * Read data from a file, folder or url and return a data object
+     *
+     * @param  string  $path
+     * @param  string  $format
+     * @throws \InvalidArgumentException If the data file(s) could not be located
+     * @throws RuntimeException
+     * @return ComPagesDataObject
+     */
+    public function createObject($path, $format = '')
+    {
+        if(!isset($this->__cache[$path]))
+        {
+            if(!parse_url($path, PHP_URL_SCHEME) == 'http')
+            {
+                //Locate the data file
+                if (!$file = $this->getObject('template.locator.factory')->locate('data://'.$path)) {
+                    throw new InvalidArgumentException(sprintf('The data file "%s" cannot be located.', $path));
+                }
+
+                $result = $this->fromPath($file);
+
+            }
+            else $result = $this->fromUrl($path, $format);
+
+            //Create the data object
+            $this->__cache[$path] = new ComPagesDataObject($result);
+        }
+
+        return $this->__cache[$path];
+    }
+
+    /**
+     * Read a data from a file or folder
+     *
+     * @param  string  $path
+     * @throws \InvalidArgumentException If the data file(s) could not be located
+     * @throws RuntimeException
+     * @return array
+     */
+    public function fromPath($path)
+    {
+        //Get the data
+        $result = array();
+        foreach((array) $path as $file)
+        {
+            $url = trim(fgets(fopen($file, 'r')));
```

Comment:
@johanjanssens 

Found a bug here or I'm just not sure what you're trying to achieve. This line opens up the file, and reads the first line. The first line is then parsed on line 66 as a URL. This causes `Cannot get content from url error`.

The following lines of code works: 

```
            // Removed line 65.
            if(parse_url($file, PHP_URL_SCHEME)) {
                $data = $this->fromUrl($file, pathinfo($file, PATHINFO_EXTENSION));
            } else {
                $data = $this->getObject('object.config.factory')->fromFile($file, false);
            }
```

Reply:
@raeldc This is by design it implements an alias feature as explained in the description. This works fine on my end, how are you testing it?